app/src/controllers/pages/typed-selectors.ts

```diff
@@ -0,0 +1,38 @@
+type LocationInfo = {
+  payload: {
+    testCasePageRoute: string | string[];
+  };
+};
+
+type State = {
+  location: LocationInfo;
+};
+
+export const locationSelector = (state: State): LocationInfo => state.location;
+
+export const payloadSelector = (state: State): LocationInfo['payload'] =>
+  locationSelector(state).payload;
+
+export const urlTestCaseSlugSelector = (state: State): string => {
+  const testCasePageRoute = payloadSelector(state).testCasePageRoute;
+
+  if (testCasePageRoute && typeof testCasePageRoute === 'string') {
+    return testCasePageRoute.split('/')[3];
+  } else if (Array.isArray(testCasePageRoute)) {
+    return testCasePageRoute?.[3];
+  }
```

Comment:
It is possible that in this branches you will get underfined. Remove `: string` from function from params. Overall please type to be derived automatically (`: string` forces typescript to believe us)

Reply:
Changed function. Now returned type is always string.